routes/likedVideos.router.js

```diff
@@ -0,0 +1,74 @@
+const express = require("express");
+const router = express.Router();
+const LikedVideo = require("../models/likedVideo.model");
+const DislikedVideo = require("../models/dislikeVideo.model");
+const User = require("../models/user.model");
+const Video = require("../models/video.model");
+const checkAuth = require("../middlewares/checkAuth");
+
+router.post("/:userId/:videoId/like", checkAuth, async (req, res) => {
+	const { userId, videoId } = req.body;
+	try {
+		const foundVideo = await Video.findById({ _id: videoId });
+		const foundUser = await User.findById({ _id: userId });
+		const foundLikedVideo = await LikedVideo.findOne({ userId });
+		const foundDislikedVideo = await DislikedVideo.findOne({ userId });
+		const checkPresent =
+			foundLikedVideo && foundLikedVideo.videos.find((el) => String(el) === String(videoId));
+		const checkInDislikedVideos =
+			foundDislikedVideo &&
```

Comment:
Instead of performing User.findById every time its required it can be handled using a paramHandler